### DISCUSSION TITLE
[QUESTION] Node has high CPU load, and PVC io performance is low

### DISCUSSION THREAD

**Discussion by qq958905365** [longhorn/longhorn#3689] [Q&A]
## Question
>I have a 6-node kubernetes with Longhorn v1.2.3 installed, I have one replica per each Longhorn volume,and each replica is scheduled on a different node

> I also deploy prometheus, postgresql, redis, kafka, rabbitmq in my cluster. 
Recently, I found my node has a very high CPU load, I using top command in the node, but I don't find any exceptions. 
![image](https://user-images.githubusercontent.com/42056766/156289053-acb513e4-527d-4a7c-b688-a5a2a7009c88.png)

> I using "iostat -x 2 10" command to search my io performance, and  I found the pvc io performance is too low:
![image](https://user-images.githubusercontent.com/42056766/156288633-a5bde7e0-f579-4641-a9fb-3779ff747b51.png)
io wait time is much than io run time, could any one help me?
## Environment

 - Longhorn version: 1.2.3
 - Kubernetes version: 1.18.4
 - Node config
   - OS type and version: centos7.6
   - CPU per node: 32
   - Memory per node: 64
   - Disk type: SSD
   - Network bandwidth and latency between the nodes: bandwidth: 5Gbps
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Tecent TKE

**Comment by derekbit** [comment]
Is the volume healthy? Is it rebuilding or running backup?

**Comment by qq958905365** [comment]
Yes，the volume is healthy,  rebuilding and backup will not spend for so long time

**Comment by derekbit** [comment]
`io wait` is pretty high. Can you check the network latencies between nodes and underlying storage iops fit your expectation?

**Comment by qq958905365** [comment]
After I adopted read-write separation for postgresql, the problem of heavy load has been solved

**Comment by derekbit** [comment]
Sounds good. Is the read-write heavy or there are lots of small IO?
For a SSD, the high io wait is somewhat weird.
